app/views/play/level/tome/Spell.coffee

```diff
@@ -157,14 +157,29 @@ module.exports = class Spell
     @view?.getSource() ? @source
 
   transpile: (source) ->
```

Comment:
Is `transpile` used in other code paths that expect the method to be synchronous?

Reply:
i think i didn't find that. all i find now is asynchronous, let me check it globally.

Reply:
I don't believe we need any changes in this file. As long as the simulation replaces the source with the C++ token all of this can happen synchronously. The C++ token can be treated like source code by our codebase.